BITCORNService/Utils/ReferralUtils.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using BITCORNService.Models;
+using BITCORNService.Utils.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace BITCORNService.Utils
+{
+    public static class ReferralUtils
+    {
+        public static void UpdateReferralSync(BitcornContext dbContext, PlatformId platformId)
+        {
+            int? userId =0;
+            switch (platformId.Platform)
+            {
+                case "twitch":
+                    userId = dbContext.UserIdentity.FirstOrDefault(u => platformId != null && u.TwitchId == platformId.Id)?.UserId;
+                    break;
+                case "twitter":
+                    userId = dbContext.UserIdentity.FirstOrDefault(u => platformId != null && u.TwitterId == platformId.Id)?.UserId;
+                    break;
+                case "discord":
+                    userId = dbContext.UserIdentity.FirstOrDefault(u => platformId != null && u.DiscordId == platformId.Id)?.UserId;
+                    break;
+                case "reddit":
+                    userId = dbContext.UserIdentity.FirstOrDefault(u => platformId != null && u.RedditId == platformId.Id)?.UserId;
+                    break;
+            }

```

Comment:
this functionality already exists in util
`int? userId = BitcornUtils.GetUserForPlatform(platformId, dbContext).Select(u => u.UserId).FirstOrDefault();`